### ISSUE TITLE
[WIP]Support for Flutter Desktop

### ISSUE THREAD

**Issue by cloudwebrtc** [flutter-webrtc/flutter-webrtc#37]
Texture widget for the embedder seems to be coming soon: https://github.com/google/flutter-desktop-embedding/issues/107#issuecomment-444272681, let's port it to the desktop embedder and let the plugin extend to mac/linux/windows.  😄

**Comment by cloudwebrtc** [comment]
We created a branch to test the desktop version, based on the patch at https://github.com/google/flutter-desktop-embedding/issues/107#issuecomment-444272681, which seems to work very well now.
If you want to test you through the following steps:

```dart
/*create a test directory*/
mkdir  flutter-desktop-webrtc-test
cd  flutter-desktop-webrtc-test

git clone https://github.com/cloudwebrtc/flutter-desktop-embedding.git
cd flutter-desktop-embedding && git checkout flutter-webrtc

/*copy pre-compiled frameworks to test directory*/
cp -rf flutter_engine_framework ../
cd flutter_engine_framework

/*link flutter tools to flutter-desktop-webrtc-test, please change to your tool path. */
ln -sf ln -sf ~/bin/flutter .

/*open xcode project to test it.*/
open flutter-desktop-embedding/example/macos/ExampleEmbedder.xcodeproj

```
## screenshots
<img width="269" height="199" src="https://raw.githubusercontent.com/cloudwebrtc/flutter-webrtc-demo/master/screenshots/flutter-webrtc-desktop-01.png"/> <img width="269" height="199" src="https://raw.githubusercontent.com/cloudwebrtc/flutter-webrtc-demo/master/screenshots//flutter-webrtc-desktop-02.png"/> <img width="269" height="199" src="https://raw.githubusercontent.com/cloudwebrtc/flutter-webrtc-demo/master/screenshots//flutter-webrtc-desktop-03.png"/>

**Comment by cloudwebrtc** [comment]
I'm so happy to complete the rendering of Texture for windows, the plugin for Linux  should be coming  soon.
![FA633AE9-376A-4311-B179-8B9C922E4E38](https://user-images.githubusercontent.com/17733746/56179629-02ed0180-6039-11e9-9425-ae079bd2e886.png)

**Comment by kangshaojun** [comment]
@cloudwebrtc Congratulations on completing the flutter video plug-in for Windows platform

**Comment by cloudwebrtc** [comment]
Currently blocked by the following issues.

Add macOS/Windows/Linux plugin tooling
https://github.com/flutter/flutter/issues/32720
https://github.com/flutter/flutter/issues/32719
https://github.com/flutter/flutter/issues/32718

Support Texture widget in macOS shell 
https://github.com/flutter/flutter/issues/30717
Support Texture widget in GLFW shell 
https://github.com/flutter/flutter/issues/30718

**Comment by cloudwebrtc** [comment]
The migration test for all desktop platforms has been passed, and after the flutter/engine Texture patch is merged, the branches can be merged into the master.
## Mac/Linux/Windows
<img width="269" height="199" src="https://user-images.githubusercontent.com/17733746/63213835-93e8fc00-c143-11e9-961a-b52139ef7971.png"/><img width="269" height="199" src="https://user-images.githubusercontent.com/17733746/63213796-2937c080-c143-11e9-876d-447735b67594.png"/><img width="269" height="199" src="https://user-images.githubusercontent.com/17733746/63213798-2ccb4780-c143-11e9-9290-a209a578c166.png"/>

**Comment by joeblew99** [comment]
This is not merged upstream into flutter engine yet ?

Just asking as I want to get this working in go-flutter also.

**Comment by cloudwebrtc** [comment]
@joeblew99 The [glfw texture](https://github.com/flutter/engine/pull/9822) and [macos texture](https://github.com/flutter/engine/pull/8507)  patch for flutter/engine, which has not yet been reviewed, currently relies on two other patches, [glad](https://github.com/flutter/buildroot/pull/275) and [flutter tools](https://github.com/flutter/flutter/pull/36818).

**Comment by joeblew99** [comment]
@cloudwebrtc thank you for the tips. Will wait for it to land upstream.

**Comment by cloudwebrtc** [comment]
The desktop plugin has moved forward one step.
https://github.com/flutter/engine/commit/0f9d88c7b635cf6d721fcc4547f51b9d60472158

**Comment by Sunbreak** [comment]
Any progress?

Since Feb 2020, Flutter Windows supports dynamic link on `vcproj`s. And we build Agora plugins with audio only:
https://github.com/woodemi/Flutter-RTM-Desktop
https://github.com/woodemi/Flutter-SDK-Desktop

Could we have `flutter-webrtc` the same way?

**Comment by cloudwebrtc** [comment]
@Sunbreak https://github.com/cloudwebrtc/flutter-webrtc/tree/glfw
There is a `GLFW` branch for `Windows/Linux`, but the texture patch upstream of flutter has not been completed.

**Comment by Sunbreak** [comment]
@cloudwebrtc According to https://github.com/flutter/engine/pull/9822#issuecomment-646028050, GLFW is deprecated.

Any further plan about flutter-webrtc?

**Comment by cloudwebrtc** [comment]
@Sunbreak `flutter-webrtc` supports all desktop on the roadmap.
looks like it needs a new patch for `Windows/Linux`. Let me take a deep breath. xD

**Comment by muriloventuroso** [comment]
Any news?

**Comment by guilhermesgb** [comment]
I'm also interested :)

(ps: thanks for your ongoing work on supporting this on many fronts)

**Comment by gedw99** [comment]
me too :) also interested

**Comment by Sunbreak** [comment]
[Linux texture support](https://github.com/flutter/engine/pull/24916) has been landed. Could we got an official Linux support?

**Comment by cloudwebrtc** [comment]
Awesome, I am already merging PR for [Windows support](https://github.com/flutter-webrtc/flutter-webrtc/pull/688), and I believe that Linux and embedded Linux will be supported soon

**Comment by gedw99** [comment]
thanks @cloudwebrtc 

Its been a long time coming from the Flutter Team, but finally !!

**Comment by dpavlak** [comment]
I'm trying to create a app to use mobile camera as a webcam.
I'm starting with flutter-webrtc-demo base code to mobile and desktop app
When i try to send mobile camera stream to windows app in a p2p call sample, the image preview on desktop app not work.

![image](https://user-images.githubusercontent.com/78757305/162085268-626a58da-ce48-4aab-9673-a1705fcac4e7.png)

the only message in console is this 
![image](https://user-images.githubusercontent.com/78757305/162085348-8c378770-bdd1-44c0-8233-f611ce424ba2.png)

can someone help pls? this is really important to me, thanks.
